### PR TITLE
hash/anchor-link scrolling in standardScroll

### LINKED ISSUE #22: Scroll to hash anchor
What is the preferred way to implement a hash/anchor-link scrolling behavior as in https://github.com/rackt/react-router/issues/2144#issuecomment-153899046 ? So far I found 2 solutions:
1. continue using history.listen (although using setTimeout in order to wait for the router behavior to evaluate seems pretty unreliable)
2. create a useHashScroll decorator function using the /utils of this repo

### PR DESCRIPTION
**Pull request by SpainTrain**
I am not certain this is the best approach (basically option 1 from https://github.com/taion/scroll-behavior/issues/22#issue-121245846), but wanted to open something up for discussion.

Changes
-------

- Check for hash and scroll to relevant element on next frame

TODO
--------

- [ ] Do not scroll to hash on POP transition
- [ ] test and account for use with other scrolling libs (e.g., smoothscroll)
- [ ] Tests

Closes #22

---
This change is [Reviewable](https://reviewable.io/reviews/taion/scroll-behavior/47)

### CODE REVIEW

File: modules/useStandardScroll.js [hunk 1]
```diff
@@ -27,9 +28,17 @@ export default function useStandardScroll(createHistory) {
...
   // `history` will invoke this listener synchronously, so `currentKey` will
   // always be defined.
-  function updateScroll({ key }) {
+  function updateScroll({ key, hash }) {
     currentKey = key
-
+    if (hash) {
+      // node may not yet be in DOM
+      requestAnimationFrame(() => {
+        const node = querySelectorAll(window, hash).pop()
```
Comment (taion) [inline modules/useStandardScroll.js]: Shouldn't this be `document.getElementById`?

**Comment by taion** [comment]
Let me think a bit about what I want to do with this library.

I need to also get https://github.com/mjackson/history/issues/163 into a good direction first as well.

File: modules/useStandardScroll.js (hunk 1 above)
Comment (SpainTrain) [inline modules/useStandardScroll.js]: Yes, it is equivalent, but since Dom helpers was already being used I used this. Can switch if native is better.

Comment (taion) [inline modules/useStandardScroll.js]: `getElementById` is better than querying by an arbitrary selector string.

Comment (SpainTrain) [inline modules/useStandardScroll.js]: reasonable :white_check_mark:

**Comment by eliseumds** [comment]
How would it affect people that use *smoothScroll* in some specific cases? It would override/cancel the animation, wouldn't it?

**Comment by taion** [comment]
I have no idea.

This PR is lacking the logic to only scroll to the hash on a new transition (e.g. we shouldn't scroll to the hash on a pop transition – we should just restore the scroll position).

**Comment by SpainTrain** [comment]
> It would override/cancel the animation, wouldn't it?

Probably.  I'm using my fork in production with react-scroll.  I will switch over to smoothscroll (which is better for my use case anyway) and see about a solution / try your suggestion.



> we shouldn't scroll to the hash on a pop transition 

:+1: This is the type of discussion I wanted to open.  I created a TODO checklist in the PR description to track outstanding items.

**Comment by taion** [comment]
By default, this should be on all actions other than `POP`, BTW.

**Comment by taion** [comment]
At this point, this change is sufficiently far behind head that it won't be directly mergeable. I'm going to close this out for now.

**Pull request closed without merging**